cmd/crank/common/version.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2023 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package common contains common functions
+package common
+
+import (
+	"context"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	"github.com/crossplane/crossplane-runtime/pkg/errors"
+)
+
+const (
+	errKubeConfig                = "failed to get kubeconfig"
+	errCreateK8sClientset        = "could not create the clientset for Kubernetes"
+	errFetchCrossplaneDeployment = "could not fetch deployments"
+)
+
+// FetchCrossplaneVersion initializes a Kubernetes client and fetches
+// and returns the version of the Crossplane deployment. If the version
+// does not have a leading 'v', it prepends it.
+func FetchCrossplaneVersion() (string, error) {
+	var version string
+	config, err := ctrl.GetConfig()
+	if err != nil {
+		return "", errors.Wrap(err, errKubeConfig)
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return "", errors.Wrap(err, errCreateK8sClientset)
+	}
+
+	deployments, err := clientset.AppsV1().Deployments("").List(context.TODO(), v1.ListOptions{
+		LabelSelector: "app=crossplane",
+	})
+	if err != nil {
+		return "", errors.Wrap(err, errFetchCrossplaneDeployment)
+	}
+
+	for _, deployment := range deployments.Items {
+		v := deployment.Labels["app.kubernetes.io/version"]
```

Comment:
What sets this label? The Helm chart?

Reply:
https://github.com/crossplane/crossplane/blob/master/cluster/charts/crossplane/templates/deployment.yaml#L8

Reply:
https://github.com/crossplane/crossplane/blob/master/cluster/charts/crossplane/templates/_helpers.tpl#L27

Reply:
I'm not sure how I feel about relying on the fact that Crossplane is deployed by Helm to get this version.

Would looking at the OCI image tag of the Crossplane container work?

Reply:
as fallback we can use the image tag - updated the code